### PR TITLE
Added dataconf as a package.

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes ready for review!

File: recipes/dataconf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - poetry
+    - python >=3.6,<4.0
+  run:
+    - pyhocon >=0.3.54,<0.4.0
+    - python >=3.6,<4.0
```
Comment (dopplershift) [inline recipes/dataconf/meta.yaml:27]: Upstream says `>=3.8`?

File: recipes/dataconf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+about:
+  home: https://github.com/zifeo/dataconf
+  summary: Simple dataclasses configuration management for Python with hocon/json/yaml/properties/env-vars/dict support.
+  license: MPL-2.0
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - MattChanTK
```
Comment (dopplershift) [inline recipes/dataconf/meta.yaml:47]: Why is this not `thewchan`?

Comment (thewchan) [reply recipes/dataconf/meta.yaml:47]: good catch, not sure why that's the default from grayskull... sometimes I forget to change it

File: recipes/dataconf/meta.yaml (hunk 1 above)
Comment (thewchan) [reply recipes/dataconf/meta.yaml:27]: I'll change this, I must have missed checking this and just used default from GS

**Comment by thewchan** [comment]
@dopplershift addressed your comments

**Review by dopplershift — APPROVED** [review]

**Pull request merged**